Description: Made build.gradle compatible to 4.1 gradle version
 Converted kts to groovy, corrected method syntax and switch to 1.3-SNAPSHOT.
Author: Saif Abdul Casim

---

--- a/core/descriptors.jvm/build.gradle.kts
+++ b/core/descriptors.jvm/build.gradle.kts
@@ -1,10 +1,10 @@
 plugins {
-    kotlin("jvm")
+    id("org.jetbrains.kotlin.jvm")
     id("jps-compatible")
 }
 
 jvmTarget = "1.6"
-javaHome = rootProject.extra["JDK_16"] as String
+javaHome = rootProject.ext["JDK_16"] as String
 
 dependencies {
     compile(project(":kotlin-annotations-jvm"))
@@ -12,15 +12,18 @@
     compile(project(":core:deserialization"))
     compile(project(":core:metadata.jvm"))
     compile(project(":core:util.runtime"))
-    compile(commonDep("javax.inject"))
+    compile(DependenciesKt.customCommonDep(project,"javax.inject"))
 }
 
 sourceSets {
-    "main" { projectDefault() }
+    "main" {
+    	java.srcDirs("src")
+		resources.srcDir("resources")
+	}
     "test" {}
 }
 
-tasks.withType<JavaCompile> {
+tasks.withType(JavaCompile) {
     sourceCompatibility = "1.6"
     targetCompatibility = "1.6"
 }
--- /dev/null
+++ b/core/descriptors.runtime/build.gradle
@@ -0,0 +1,40 @@
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+
+plugins {
+    id("org.jetbrains.kotlin.jvm")
+}
+
+dependencies {
+    compileOnly(project(":core:util.runtime"))
+    compileOnly(project(":core:descriptors"))
+    compileOnly(project(":core:descriptors.jvm"))
+
+    //testCompile(DependenciesKt.customProjectTests(it,":compiler:tests-common"))
+    //testCompile(DependenciesKt.customProjectTests(it,":generators:test-generator"))
+}
+
+sourceSets {
+    "main" {
+    	java.srcDirs("src")
+		resources.srcDir("resources")
+	}
+    "test" {
+    	java.srcDirs("test", "tests")
+    }
+}
+
+tasks.withType(JavaCompile).getByName("compileJava") { // dont need to add it to tasks again as this is groovy
+    sourceCompatibility = "1.6"
+    targetCompatibility = "1.6"
+}
+
+tasks.withType(KotlinCompile).getByName("compileKotlin") { // dont need to add it to tasks again as this is groovy
+    kotlinOptions.jvmTarget = "1.6"
+    kotlinOptions.jdkHome = rootProject.ext["JDK_16"] as String
+}
+
+//def generateTests = CommonUtilKt.customGenerator(project,"org.jetbrains.kotlin.generators.tests.GenerateRuntimeDescriptorTestsKt",null) //dont probably need this
+
+TasksKt.customProjectTest(project,"test"){
+    it.workingDir = rootDir
+}
--- a/core/descriptors.runtime/build.gradle.kts
+++ /dev/null
@@ -1,35 +0,0 @@
-import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
-
-plugins {
-    kotlin("jvm")
-}
-
-dependencies {
-    compileOnly(project(":core:util.runtime"))
-    compileOnly(project(":core:descriptors"))
-    compileOnly(project(":core:descriptors.jvm"))
-
-    testCompile(projectTests(":compiler:tests-common"))
-    testCompile(projectTests(":generators:test-generator"))
-}
-
-sourceSets {
-    "main" { projectDefault() }
-    "test" { projectDefault() }
-}
-
-val compileJava by tasks.getting(JavaCompile::class) {
-    sourceCompatibility = "1.6"
-    targetCompatibility = "1.6"
-}
-
-val compileKotlin by tasks.getting(KotlinCompile::class) {
-    kotlinOptions.jvmTarget = "1.6"
-    kotlinOptions.jdkHome = rootProject.extra["JDK_16"] as String
-}
-
-val generateTests by generator("org.jetbrains.kotlin.generators.tests.GenerateRuntimeDescriptorTestsKt")
-
-projectTest {
-    workingDir = rootDir
-}
--- a/core/metadata.jvm/build.gradle.kts
+++ b/core/metadata.jvm/build.gradle.kts
@@ -1,21 +1,24 @@
 plugins {
-    kotlin("jvm")
+    id("org.jetbrains.kotlin.jvm")
     id("jps-compatible")
 }
 
 jvmTarget = "1.6"
-javaHome = rootProject.extra["JDK_16"] as String
+javaHome = rootProject.ext["JDK_16"] as String
 
 dependencies {
     compile(project(":core:metadata"))
 }
 
 sourceSets {
-    "main" { projectDefault() }
+    "main" {
+    	java.srcDirs("src")
+		resources.srcDir("resources")
+	}
     "test" {}
 }
 
-tasks.withType<JavaCompile> {
+tasks.withType(JavaCompile) {
     sourceCompatibility = "1.6"
     targetCompatibility = "1.6"
 }
--- a/core/util.runtime/src/org/jetbrains/kotlin/config/KotlinCompilerVersion.java
+++ b/core/util.runtime/src/org/jetbrains/kotlin/config/KotlinCompilerVersion.java
@@ -21,7 +21,7 @@
 public class KotlinCompilerVersion {
     // The value of this constant is generated by the build script
     // DON'T MODIFY IT
-    public static final String VERSION = "@snapshot@";
+    public static final String VERSION = "1.3-SNAPSHOT";
 
     // True if the latest stable language version supported by this compiler has not yet been released.
     // Binaries produced by this compiler with that language version (or any future language version) are going to be marked
